deployments/deploy-tests.ps1

```diff
@@ -1,99 +1,122 @@
     Import-Module .\powerform\pf-deployment-context\pf-deployment-context.psd1
+    Import-Module .\azure\pf-azure-context\pf-azure-context.psd1
+    Import-Module .\azure\resources\pf-key-vault\pf-key-vault.psd1
+    Import-Module .\azure\resources\pf-log-analytics\pf-log-analytics.psd1
 
-    Get-Command -Module log-analytics
-    Get-Command -Module key-vault
-    Get-Command -Module pf-deployment-context
-    Get-Command -Module az-cloud-context
+    #shows what functions are exported from each module
+    Get-Command -Module pf-key-vault
+    Get-Command -Module pf-log-analytics
+    Get-Command -Module pf-deployment-context 
+    Get-Command -Module pf-azure-context
 
     ##############################################################################################################
     #
     #               Deployment Context can drive naming defaults 
     #
     #############################################################################################################
 
-    #Still need to build this. The thought behind this is that a user wants to deploy resources out with best practices/enterprise standards
+    #The thought behind this is that a user wants to deploy resources out with best practices/enterprise standards
     #baked into the module.  By defining the deploy context parameters, the names will be created, therefore it knows
     #the RG, Sub, Resources names.  All options are going to be the defaults, so just build the definitions and deploy
 
-    #$deployContext = New-PfDeploymentContext -CompanyAbbreviation "MYAEA" -GroupAbbreviation "KKZH" -TeamAbbreviation "ADM" -EnvironmentLetter "D"
+    New-PfDeploymentContext
+
+    Set-PfAzureContext -CompanyAbbreviation "MYAEA" -GroupAbbreviation "KKZH" -TeamAbbreviation "ADM" -EnvironmentLetter "D" -Region "CentralUs"
+
+    #name, az context built from deployment context params above
+    $kv = New-PfKeyVaultBuild
 
     #name, az context built from deployment context params above
-    #$kv = New-KeyVaultDefinition
+    $la = New-PfLogAnalyticsBuild
+
+    Deploy-PfDeploymentContext
+
+     ##############################################################################################################
+    #
+    #               Switch Az Contexts in the middle of deployment context
+    #
+    #############################################################################################################
+
+    #Shows you can build a keyvault for the admin team, then switch context to the "DM" team and build a LA
+
+    New-PfDeploymentContext
+
+    Set-PfAzureContext -CompanyAbbreviation "MYAEA" -GroupAbbreviation "KKZH" -TeamAbbreviation "ADM" -EnvironmentLetter "D" -Region "CentralUs"
 
     #name, az context built from deployment context params above
-    #$la = New-LogAnalyticsDefinition
+    $kv = New-PfKeyVaultBuild
+
+    Set-PfAzureContext -CompanyAbbreviation "MYAEA" -GroupAbbreviation "KKZH" -TeamAbbreviation "DM" -EnvironmentLetter "D" -Region "CentralUs"
```

Comment:
could this support only overriding the ones wanting to be changed from existing context, so "leave everything the same, but change the team i'm working on", since the global def already has all the things set.  default values for parms from the global def?  if not set would have to error out...

Reply:
I show how I am doing this in one of the deploy tests with you having to still supply all params.  I guess we could create a "Change-PfAzureContext" function where you can supply one or all params to override...  May be something to look into.

Reply:
We have "Update-PfAzureContext" now implemented